src/core/server/http/router/route.ts

```diff
@@ -116,13 +116,14 @@ export interface RouteConfigOptionsBody {
  */
 export interface RouteConfigOptions<Method extends RouteMethod> {
   /**
-   * A flag shows that authentication for a route:
-   * `enabled`  when true
-   * `disabled` when false
+   * Defines authentication mode for a route:
+   * - true. A user has to have valid credentials to access a resource
+   * - false. A user can access a resource without any credentials.
+   * - 'optional'. A user can access a resource if has valid credentials or no credentials at all.
```

Comment:
Could we add a `@remarks` section that explains why/when `'optional'` is useful?